src/commands/LettuceMeetVisualizer.ts

```diff
@@ -0,0 +1,292 @@
+      const end = startTime + timeBlock * 60 * 1000;
+      for (let i = startTime; i < end; i += LETTUCE_MEET_INTERVAL * 60 * 1000) {
+        const currentNumber = availabilityMap.get(i)?.numAvailable || 0;
+        minimumAvailable = Math.min(minimumAvailable, currentNumber);
+      }
+
+      // Returns the start time and the minimum number of people available.
+      return [startTime, { numAvailable: minimumAvailable, peopleAvailable: availabilityData.peopleAvailable }];
+    });
+
+    bestTimes.sort((a, b) => b[1].numAvailable - a[1].numAvailable);
+    return bestTimes.map(([startTime, availablityData]) => [startTime, availablityData]);
+  }
+
+  /**
+   * Gets the availability map for a given board.
+   * @param boardData board data
+   * @param filterBy emails or names of the users to filter by
+   * @returns availability map for the given board
+   */
+  private async getAvailabilityMap(boardData: BoardData, filterBy?: string[]): Promise<Map<number, AvailabilityData>> {
+    const pollResponses = boardData.data.event.pollResponses;
+    const filteredPollResponses = pollResponses.filter((pollResponse) => {
+      if (!filterBy) {
+        return true;
+      }
+
+      return filterBy.includes(pollResponse.user.email) || filterBy.includes(pollResponse.user.name);
+    });
+
+    // Key is the start time for the interval, value is the number of people available.
+    const availabilityMap = new Map<number, AvailabilityData>();
+    filteredPollResponses.forEach((pollResponse) => {
+      pollResponse.availabilities.forEach((availability) => {
+        // Parses start and end date into a Date object
+        // Substring to remove the 'Z' at the end of the string.
+        const start = new Date(availability.start.substring(0, availability.start.length - 1));
+        const end = new Date(availability.end.substring(0, availability.end.length - 1));
+
+        // Loops through each LETTUCE_MEET_INTERVAL minute interval and increments the number of people available.
+        for (let i = start.getTime(); i < end.getTime(); i += LETTUCE_MEET_INTERVAL * 60 * 1000) {
+          const numAvailable = (availabilityMap.get(i)?.numAvailable || 0) + 1;
+          const peopleAvailable = availabilityMap.get(i)?.peopleAvailable || [];
+
+          peopleAvailable.push(pollResponse.user.name);
+          availabilityMap.set(i, { numAvailable, peopleAvailable });
+        }
+      });
+    });
+
+    return availabilityMap;
+  }
+
+  /**
+   * Returns the user availability a specific user.
+   * @param boardCode code of the lettuce meet board
+   * @param userOptions the user's name or email
+   * @returns user's availability or null if the user is not found
+   */
+  private async getUserAvailability(boardCode: string,
+    { name, email }: {
+      name?: string | null,
+      email?: string | null
+    }): Promise<UserAvailability[] | null> {
+    console.log('Getting user availability...');
```

Comment:
We should be using Logger stuff instead of console.log, this is just so logs will be stored persistently in the EC2 instance. Also, for logs its nice to add where the log is from (/lettucemeet: Getting user availability for board ___)